### PR TITLE
Remove auth

### LINKED ISSUE #94: Remove auth
Having users and auth in the Phase 1 prototype isn't needed, and it's been tripping us up on a few occasions, so let's just get rid of it.

### PR DESCRIPTION
**Pull request by waxlamp**
This PR removes auth stuff, including permissions checking, ownership of resources, and the concept of a user.

This simplifies testing and setup, and should allow for easier access to the codebase for smaller-time contributors like myself.

I need some good reviews of this to make sure I'm not breaking anything. In particular, I'm not sure if the populate script is working properly (I think it is, but I'm not sure), and I had to add an empty list `permission_classes` attribute to the Job model, even though it doesn't seem like I should need to do that.

Closes #94

### CODE REVIEW

File: atlascope/core/management/populate/investigations.json
Comment (annehaley) [inline atlascope/core/management/populate/investigations.json:3]: We should probably change the names of the datasets we create, and perhaps only create one.

File: atlascope/core/migrations/0009_auto_20220302_1642.py
Comment (annehaley) [inline atlascope/core/migrations/0009_auto_20220302_1642.py]: Can you change the name of this migration file to be more specific? Like `remove_user_fields`

File: atlascope/urls.py
Comment (annehaley) [inline atlascope/urls.py:36]: Perhaps we should rip out `oauth` entirely, since this is about simplification. At this point, we can remove `oauth2_provider` and `guardian` from `INSTALLED_APPS` in our settings so we don't end up migrating the db to support these things. Then we can also remove them from our dependencies.

File: atlascope/core/management/commands/populate.py
Comment (annehaley) [inline atlascope/core/management/commands/populate.py:75]: We need to update lines 8-9 in the README; the command invocation has changed and `createsuperuser` is now required if you want to use the admin console.

File: atlascope/core/management/commands/populate.py
Comment (annehaley) [inline atlascope/core/management/commands/populate.py]: we also don't need `DEFAULT_CLIENT_URI`

File: atlascope/core/management/commands/populate.py
Comment (annehaley) [inline atlascope/core/management/commands/populate.py]: remove

File: atlascope/core/models/investigation.py
Comment (annehaley) [inline atlascope/core/models/investigation.py]: we probably don't need the `InvestigationSerializer` vs `InvestigationDetailSerializer` distinction anymore, just use one serializer here

File: atlascope/core/rest/endpoints/job_endpoints.py
Comment (annehaley) [inline atlascope/core/rest/endpoints/job_endpoints.py]: You'll actually need this in the other viewsets as well. The reason that you were only getting 401s for job endpoints is because those tests are the only "write" endpoints we test. If we add a test for creating a dataset, the same thing would happen to that test without this line in `dataset_endpoints.py`

File: atlascope/core/rest/endpoints/pin_endpoints.py
Comment (annehaley) [inline atlascope/core/rest/endpoints/pin_endpoints.py]: You can probably do the same as you have done on other viewsets and simply specify `queryset = Pin.objects.all().order_by('id')` instead of `get_queryset`

**Review by annehaley — CHANGES_REQUESTED** [review]
Sad to see so much code go, but it is looking a lot slimmer now. :smiling_face_with_tear: 
It appears there are a lot of unused imports now (see CI lint test). I didn't carefully read any of the changes in `web` (I defer to @naglepuff for those), but I did test using the frontend and it works fine without auth. The populate script works fine as well.

File: atlascope/core/migrations/0009_auto_20220302_1642.py
Comment (waxlamp) [reply atlascope/core/migrations/0009_auto_20220302_1642.py]: See 029dc08eb333596803b65493801f116dbe068b92.

File: atlascope/core/management/commands/populate.py
Comment (waxlamp) [reply atlascope/core/management/commands/populate.py:75]: See 40f93d89f0845ea254775bca52d8d414f3bbe3a9.

File: atlascope/core/management/commands/populate.py
Comment (waxlamp) [reply atlascope/core/management/commands/populate.py]: See b40c3f7ff058b2170bc7beb669320f731b72c3b8.

File: atlascope/core/management/commands/populate.py
Comment (waxlamp) [reply atlascope/core/management/commands/populate.py]: I must have removed this between my PR going live and my seeing this comment; not sure which commit that was.

File: atlascope/core/models/investigation.py
Comment (waxlamp) [reply atlascope/core/models/investigation.py]: See 74c829c5ae34fabd6b41d42a95591adb9ec0cd71.

File: atlascope/core/rest/endpoints/job_endpoints.py
Comment (waxlamp) [reply atlascope/core/rest/endpoints/job_endpoints.py]: Instead of doing that, I figured out the configuration option to disable permission management in all routes: fdedfe99cb8acab7b5e40a4e79364635b799d0a5.

File: atlascope/core/rest/endpoints/pin_endpoints.py
Comment (waxlamp) [reply atlascope/core/rest/endpoints/pin_endpoints.py]: See 0458668a7bf17eac5a1eb723fcda1936121a3966.

File: atlascope/core/management/populate/investigations.json
Comment (waxlamp) [reply atlascope/core/management/populate/investigations.json:3]: Once @mcovalt's examples land in the codebase a bit more solidly, I think it would make sense to create a couple of investigations featuring the speckled test data and then a more realistic one, but I can take care of that in a future PR.

File: atlascope/urls.py
Comment (waxlamp) [reply atlascope/urls.py:36]: I agree in principle but also it won't do any harm to leave this stuff in place. The goal, more than simplification, is an easier or simplified developer experience. I looked into ripping these things out by the root and it seems like more trouble than it's worth (other than guardian, which I will do; stand by).

Comment (waxlamp) [reply atlascope/urls.py:36]: See 142f37f7e86129df41dedef41bda1f42b6727b92.

**Comment by waxlamp** [comment]
> Sad to see so much code go, but it is looking a lot slimmer now. smiling_face_with_tear

Like the Klingons, I do not consider the occasion of code deletion to be one of mourning, but rather celebration, as it rises to the ranks of the honored deleted in [Suto'vo'qor](https://memory-alpha.fandom.com/wiki/Sto-vo-kor). If you prefer human reflections, [Shakespeare](https://www.poetryfoundation.org/poems/45098/sonnet-71-no-longer-mourn-for-me-when-i-am-dead) is surprisingly relevant 😮 

But either way, this is ready for another look 😃 

@naglepuff, still need your eyes on the frontend changes.

**Review by annehaley — APPROVED** [review]

**Review by naglepuff — APPROVED** [review]
Looks good to me! Client side code changes were straightforward and I was able to do some testing with the front end app and nothing appears to be broken. Thanks for doing this. I guess we don't technically need the `girder-oauth-client` library anymore but I think you mentioned in another comment that there's not really any harm in leaving those kinds of things in for now.

**Comment by waxlamp** [comment]
> Looks good to me! Client side code changes were straightforward and I was able to do some testing with the front end app and nothing appears to be broken. Thanks for doing this. I guess we don't technically need the `girder-oauth-client` library anymore but I think you mentioned in another comment that there's not really any harm in leaving those kinds of things in for now.

Frontend libraries are easier to remove than the backend ones, just because a lot of the default Django settings refer to those. I'll rip out girder-oauth-client.

**Pull request merged**
